src/libraries/Microsoft.PowerFx.Core/Entities/External/ServiceCapabilities2.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+#pragma warning disable SA1117
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    public class ServiceCapabilities2
+    {
+        public readonly SortRestriction2 SortRestriction;
+
+        public readonly FilterRestriction2 FilterRestriction;
+
+        public readonly SelectionRestriction2 SelectionRestriction;
+
+        public readonly GroupRestriction2 GroupRestriction;
+
+        public readonly IEnumerable<string> FilterFunctions;
+
+        public readonly IEnumerable<string> FilterSupportedFunctions;
+
+        public readonly PagingCapabilities2 PagingCapabilities;
+
+        public IReadOnlyCollection<KeyValuePair<string, ColumnCapabilitiesBase2>> ColumnsCapabilities;
+
+        public readonly bool SupportsDataverseOffline;
+
+        public readonly bool SupportsRecordPermission;
+
+        public readonly string TableName;
+
+        public readonly bool IsReadOnly;
+
+        public readonly FormulaType RecordType;
```

Comment:
we ought to be able to avoid this. 
This is class is getting passed into RecordType ctor.

Reply:
This is used in 
- public LazyTypeProvider(TableParameters tableCapabilities)
- public InternalTableCapabilities(TableParameters serviceCapabilities2)